### DISCUSSION TITLE
SendSignedTransaction: Option to switch off polling for transaction receipt

### DISCUSSION THREAD

**Discussion by emcveigh** [ethereum/web3.js#6796] [Q&A]
We are currently using sendSignedTransaction and exit the call when the transactionHash promi event is emitted.  We monitor transactions and events for the resulting sendRawTransaction through a different process and therefore it is not important for us to wait on the transactionReceipt that is returned from sendSignedTransaction.  We have noticed that even though we stop listening after the transactionHash promi event has been emitted that the sendSignedTransaction function continues to poll for transaction receipts and this in turn uses (for us) alot of unneccessary rpc calls to our provider.  We have attempted to configure this and have managed to slow the number of calls, but would like to know if there are any plans of adding an option to this function to NOT poll for transaction receipts??

Many thanks for you help!

**Comment by arijoon** [comment]
Unfortunately this is a major issue for us as well and eats a lot of rpc calls. It would be ideal if the developers change the interface of this call all together and force the user to poll for the receipt, so they are aware of the cost it could bear. If you're on v4, can easily avoid the polling call by using the underlying core functions following the example bellow:

Assuming `signedTx` is the result of `account.signTransaction`
```ts
import { ethRpcMethods } from 'web3-rpc-methods'

const signedTransactionFormattedHex = web3.utils.format(
  { format: 'bytes' },
  signedTx,
  ETH_DATA_FORMAT,
)

// The standard web3.eth.sendSignedTransaction does so much useless work
// and has a bug with unhandled promise rejections: https://github.com/web3/web3.js/issues/6900
const resultTxHash = await ethRpcMethods.sendRawTransaction(
  web3.requestManager,
  signedTransactionFormattedHex,
)
```

I'm yet to find a way to do this in web3v1

**Reply by emcveigh** [reply]
Thank you @arijoon for the reply - this is indeed exactly what we had to do so that we could proceed without using as many rpc calls.

**Comment by TatyOko28** [comment]
Currently, Web3.js does not provide an option to disable polling for transaction receipts in sendSignedTransaction(). However, you can:
 Stop polling manually by exiting after the transactionHash event.
 Use eth.sendRawTransaction() instead to send the transaction without Web3.js' built-in receipt polling.
 Adjust polling settings in the Web3.js provider configuration to reduce RPC usage.
